src/it/resources/com/google/checkstyle/test/chapter4formatting/rule412nonemptyblocks/InputTryCatchIfElse.java

```diff
@@ -0,0 +1,59 @@
+package com.google.checkstyle.test.chapter4formatting.rule412nonemptyblocks;
+
+/** some javadoc. */
+public class InputTryCatchIfElse {
+
+  @interface TesterAnnotation {} //ok
+
+  void foo() throws Exception {
+    int a = 90;
+
+    if (a == 1) {
+    } else {} // violation ''}' at column 13 should be alone on a line.'
+
+    if (a == 1) {
+    } else { }
+    // 2 violations above:
+    //  'Empty blocks should have no spaces.'
+    //  ''}' at column 14 should be alone on a line.'
+
+    if (a == 45) {}
+
+    if (a == 9) {} else {} // violation ''}' at column 26 should be alone on a line.'
```

Comment:
false-negative for `if (a == 9) {}`, it should follow KR style. The following example is ok though:

https://github.com/checkstyle/checkstyle/blob/c0cbda5c074359393ad2d5e2f05a3bfbfa2b6358/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule412nonemptyblocks/InputTryCatchIfElse.java#L20

Reply:
I don't think this is false-positive.
This multi-block structure, so `{}` is not ok.

Reply:
`if (a == 45)` so it cannot be counted in multi-block structure